aspnetcore/signalr/configuration.md

```diff
@@ -0,0 +1,129 @@
+---
+title: ASP.NET Core SignalR configuration
+author: rachelappel
+description: Configure ASP.NET Core SignalR Apps
+manager: wpickett
+monikerRange: '>= aspnetcore-2.1'
+ms.author: rachelap
+ms.date: 06/30/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/configuration
+---
+
+# ASP.NET Core SignalR configuration
+
+## JSON/MessagePack serialization options
+
+ASP.NET Core SignalR supports two protocols for encoding messages, [JSON](https://www.json.org/) and [MessagePack](https://msgpack.org/index.html). Each protocol has serialization configuration options.
+
+JSON serialization can be configured on both the server and .NET Client. The following is an example of server-side configuration:
+
+[!code-csharp[Startup config](configuration/sample/config-startup.cs?range=1-5)]
+
+The preceding code adds add `AddJsonHubProtocol` to the [AddSignalR](/dotnet/api/microsoft.extensions.dependencyinjection.signalrdependencyinjectionextensions.addsignalr#Microsoft_Extensions_DependencyInjection_SignalRDependencyInjectionExtensions_AddSignalR_Microsoft_Extensions_DependencyInjection_IServiceCollection_) call in [ConfigureServices](/dotnet/api/microsoft.aspnetcore.hosting.istartup.configureservices#Microsoft_AspNetCore_Hosting_IStartup_ConfigureServices_Microsoft_Extensions_DependencyInjection_IServiceCollection_). Then the code provides a delegate to the `AddJsonHubProtocol` method. The `options` object received by that delegate has a `PayloadSerializedSettings`. The `PayloadSerializedSettings` is a JSON.NET [JsonSerializerSettings](/dotnet/api/microsoft.aspnetcore.mvc.formatters.jsonserializersettingsprovider object that can be used to configure serialization of arguments and return values. See the [JSON.NET Documentation](https://www.newtonsoft.com/json/help/html/Introduction.htm) for more details.
```

Comment:
The wording here seems a bit awkward. What about moving it above the example:

JSON serialization can be configured on the server using the `AddJsonProtocol` extension method. The `AddJsonProtocol` method takes a delegate that receives an `options` object. The `PayloadSerializerSettings` property on that object is a JSON.NET JsonSerializerSettings object that can be used to configure serialization of arguments and return values. See the JSON.NET Documentation for more details.

As an example, you can configure the serializer to use `PascalCase` names instead of the default `camelCase` names using the following code:

[example]

In the .NET client, the same `AddJsonHubProtocol` extension method exists on HubConnectionBuilder. The `Microsoft.Extensions.DependencyInjection` namespace must be imported to resolve the extension method:

[example]